Updates

DerAndereAndi opened the pull request:
- Update zeroconf reference to include latest Windows fixes that might help us as well
- Allow setting site voltage from evcc, so calculations of minPower, maxPower, currentPower etc. could be changed based on the systems local grid setup
- Improve log output to exclude binary data
- Improve 10m timeout connection close workaround for Elli
- Switch machine-id library

DerAndereAndi committed:
Update zeroconf

Update zeroconf reference to include latest Windows fixes that might help us as well

DerAndereAndi committed:
Allow setting voltage from evcc

DerAndereAndi committed:
Fix crash

Fixes concurrent map read and map write issue

DerAndereAndi committed:
Upgrade zeroconf to v2.2.0

DerAndereAndi committed:
Do not require all values to process some of it

DerAndereAndi commented:
This is ready for merging

andig commented on communication/connectioncontroller.go:
```diff
@@ -44,8 +47,8 @@ func NewConnectionController(log util.Logger, conn ship.Conn, local spine.Device
 			Measurements: EVMeasurementsType{
-				Current: make(map[uint]float64),
-				Power:   make(map[uint]float64),
+				Current: sync.Map{},
```
Brauchts nicht

andig replied:
Durch diese Änderung kommen die ganzen Typecasts rein. Ist das notwendig?

DerAndereAndi replied:
Ja, durch den Zugriff aus mehreren Threads (schreiben und lesen gleichzeitig) ging es am schnellsten mit der sync Map. Wie gesagt, ich möchte es hier nur lauffähig machen und nicht unnötig Zeit reinstecken.

DerAndereAndi committed:
Remove unnecessary code

DerAndereAndi committed:
Improve Elli keepalive hack

Only send the keep alive SPINE message if there really was silence since 8 minutes

DerAndereAndi committed:
Don’t print the leading non ascii byte

DerAndereAndi committed:
Reset EVData on EV connection

DerAndereAndi committed:
Fix Elli workaround for 10 minutes timeout

The timeouts seem to happen exactly 10 minutes after the last received Spine message. If we are sending heartbeats in that timespan, the timeout will still happen. So for the workaround we now only consider received messages

DerAndereAndi committed:
Switch machine-id library

DerAndereAndi commented:
@andig Wie soll es hier weiter gehen?

DerAndereAndi committed:
Add method for providing machineID externally

This helps evcc as it has an additional way to provide a machineID via its configuration file

andig commented on go.mod:
```diff
@@ -3,16 +3,17 @@ module github.com/evcc-io/eebus
-	github.com/libp2p/zeroconf/v2 v2.0.0-20220623102032-af1f1d3ada85
+	github.com/libp2p/zeroconf/v2 v2.2.0
 	github.com/mitchellh/mapstructure v1.4.1
+	github.com/panta/machineid v1.0.1
```
@DerAndereAndi das ist noch alt? Mal `go mod tidy`?

DerAndereAndi replied:
Nein, das ist nicht alt. Warum meinst du?

andig replied:
Ich musste auch denisbrodbeck zurück switchen. Beides würde entfallen wenn wir die ID von evcc rein reichen lassen.

andig commented on ship/uniqueid.go:
```diff
@@ -1,9 +1,10 @@
 	"fmt"
 
-	"github.com/denisbrodbeck/machineid"
+	"github.com/panta/machineid"
```
Das muss noch raus.

DerAndereAndi replied:
Ich habe die alte Funktion als Fallback absichtlich drin gelassen. Siehe die Einbindung hier: https://github.com/evcc-io/evcc/pull/4207/files#diff-57a13a6ebe2afaf22c517ea8970440f3c1a972298fcb3ff91a87341ec6c2fc41

andig commented on ship/uniqueid.go:
```diff
@@ -15,3 +16,14 @@ func UniqueID(prefix, appID string) (string, error) {
+// UniqueID creates ship ID with given prefix and salted by provided protectedID
+// The protectedID is basically the same as `machineid.ProtectedID(appId)` above
+// but provided by the system using this library
+func UniqueIDWithProtectedID(prefix, protectedID string) (string, error) {
```
Könntest Du stattdessen eine Funktion reinreichen lassen, so dass evcc die zu verwendende machine-id bestimmen kann und damit den evcc Standard nutzen?

DerAndereAndi replied:
Ich reiche den von evcc generierten Wert rein. Siehe hier: https://github.com/evcc-io/evcc/pull/4207/files#diff-57a13a6ebe2afaf22c517ea8970440f3c1a972298fcb3ff91a87341ec6c2fc41

andig replied:
Ich würde vorschlagen den Fallback rauszunehmen. Die id muss von evcc kommen; das stellt sicher dass sie da in der Config auch persistiert ist.

DerAndereAndi replied:
Alles klar. Ist erledigt.

DerAndereAndi committed:
Remove old UniqueID generation
